spec/services/metrics/merge_time/per_user_project_spec.rb

```diff
@@ -0,0 +1,163 @@
+        let!(:pull_request) do
+          create(:pull_request,
+                 opened_at: Time.zone.now,
+                 merged_at: Time.zone.now + 30.minutes,
+                 project_id: user_project.project_id,
+                 owner_id: user_project.user_id)
+        end
+
+        it 'generates a metric with value expressed as decimal equal to 30 minutes' do
+          described_class.call
+          expect(Metric.last.value.seconds).to eq(30.minutes)
+        end
+
+        it 'generates only that metric' do
+          expect { described_class.call }.to change { Metric.count }.from(0).to(1)
+        end
+      end
+
+      context 'when calculating the merge time value' do
+        context 'and a user made more than one pull request' do
+          let!(:pull_request) do
+            create(:pull_request,
+                   opened_at: Time.zone.now,
+                   merged_at: Time.zone.now + 30.minutes,
+                   project_id: user_project.project_id,
+                   owner_id: user_project.user_id)
+          end
+
+          let!(:second_pull_request) do
+            create(:pull_request,
+                   opened_at: Time.zone.now,
+                   merged_at: Time.zone.now + 45.minutes,
+                   project_id: user_project.project_id,
+                   owner_id: user_project.user_id)
+          end
+
+          let!(:third_pull_request) do
+            create(:pull_request,
+                   opened_at: Time.zone.now,
+                   merged_at: Time.zone.now + 2.hours,
+                   project_id: user_project.project_id,
+                   owner_id: user_project.user_id)
+          end
+
+          it 'creates just one metric' do
+            expect { described_class.call }.to change { Metric.count }.from(0).to(1)
+          end
+        end
+      end
+    end
+
+    context 'when a user has pull requests in more than one project' do
+      let(:same_user_for_second_project) { create(:users_project, user_id: user_project.user_id) }
+
+      let!(:pull_request) do
+        create(:pull_request,
+               opened_at: Time.zone.now,
+               merged_at: Time.zone.now + 30.minutes,
+               project_id: user_project.project_id,
+               owner_id: user_project.user_id)
+      end
+
+      let!(:second_pull_request) do
+        create(:pull_request,
+               opened_at: Time.zone.now,
+               merged_at: Time.zone.now + 2.hours,
+               project_id: same_user_for_second_project.project_id,
+               owner_id: user_project.user_id)
+      end
+
+      before { described_class.call }
+
+      it 'it generates the metric for the first project' do
+        expect(Metric.first.value.seconds).to eq(30.minutes)
+      end
+
+      it 'it generates the metric for the second project' do
+        expect(Metric.second.value.seconds).to eq(2.hours)
+      end
+    end
+
+    context 'when has a pull request in another project' do
```

Comment:
Should this be `when another user has a pull request in the same project`?

Reply:
Is the same user in different project. In the `third_pull_request` is where i changed the project ID and not the user ID.